docs/stopping_events/syscalls.md

```diff
@@ -0,0 +1,206 @@
+        print("exiting open")
+        t.syscall_return = 0x0
+
+    handler = d.handle_syscall(syscall="open", on_enter=on_enter_open, on_exit=on_exit_open)
+    ```
+
+!!! ABSTRACT "Example of synchronous syscall handling"
+    ```python
+    from libdebug import debugger
+
+    d = debugger("./test_program")
+    d.run()
+
+    handler = d.handle_syscall(syscall="open")
+    d.cont()
+
+    if handler.hit_on_enter(d):
+        print("open syscall was entered")
+    elif handler.hit_on_exit(d):
+        print("open syscall was exited")
+    ```
+
+    The script above will print "open syscall was entered".
+
+## :octicons-number-24: Resolution of Syscall Numbers
+Syscall handlers can be created with the identifier number of the syscall or by the syscall's common name. In the second case, syscall names are resolved from a definition list for Linux syscalls on the target architecture. The list is fetched from [mebeim's syscall table](https://syscalls.mebeim.net). We thank him for hosting such a precious resource. Once downloaded, the list cached internally. 
+
+## :material-arrow-decision: Hijacking
+When hijacking a syscall, the user can provide an alternative syscall to be executed in place of the original one. Internally, the hijack is implemented by registering a handler for the syscall and replacing the syscall number with the new one.
+
+!!! ABSTRACT "Function Signature"
+    ```python
+    d.hijack_syscall(original_syscall, new_syscall, recursive=False, **kwargs) 
+    ```
+
+**Parameters**:
+
+| Argument | Type | Description |
+| --- | --- | --- |
+| `original_syscall` | `int` \| `str` | The syscall number or name to be hijacked. If set to `"*"` or `"all"`, all syscalls will be hijacked. |
+| `new_syscall` | `int` \| `str` | The syscall number or name to be executed instead. |
+| `recursive` | `bool` | If set to `True`, the handler's callback will be executed even if the syscall was triggered by a hijack. |
+| `**kwargs` | `(int, optional)` | Additional arguments to be passed to the new syscall. |
+
+**Returns**:
+
+| Return | Type | Description |
+| --- | --- | --- |
+| `SyscallHandler` | [SyscallHandler](../../from_pydoc/generated/data/syscall_handler) | The handler object created. |
+
+
+!!! ABSTRACT "Example of hijacking a syscall"
+    <div class="grid cards" markdown>
+
+    ```C
+    #include <unistd.h>
+
+    char secretBuffer[32] = "The password is 12345678";
+
+    int main(int argc, char** argv)
+    {
+        [...]
+
+        read(0, secretBuffer, 31);
+        
+        [...]
+        return 0;
+    }
+    ```
+
+    ```python
+    from libdebug import debugger
+
+    d = debugger("./test_program")
+    d.run()
+
+    handler = d.hijack_syscall("read", "write")
+    
+    d.cont()
+    d.wait()
+
+    out = pipe.recvline()
+    print(out.decode())
+    ```
+
+    </div>
+    In this case, the secret will be leaked to the standard output instead of being overwritten with content from the standard input.
```

Comment:
Well, not really because you would still be writing to STDIN in this example

Reply:
It works anyway in Linux, because they are the same symbolic link underneath (/dev/pts/...). This example was crafted this way for this reason.

Reply:
Fair